### PR TITLE
Add max zoom default

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/geoviews/issues/744

- [] After this gets merged, update https://github.com/holoviz/geoviews/pull/741

image

Unfortunately, I couldn't target OSM specifically; all the combinations I tried:
```python
options.TilePlot.OSM = Options('style', max_zoom=10)
options.TilePlot.Osm = Options('style', max_zoom=10)
options.WMTS.Osm = Options('style', max_zoom=10)
options.WMTS.OSM = Options('style', max_zoom=10)
options.Wmts.OSM = Options('style', max_zoom=10)
options.Wmts.Osm = Options('style', max_zoom=10)
options.TileSources.OSM = Options('style', max_zoom=10)
options.Tile_Sources.OSM = Options('style', max_zoom=10)
options.Tile_Sources.OSM = Options('style', max_zoom=10)
options.TileSources.Osm = Options('style', max_zoom=10)

```

### CODE REVIEW

**Comment by hoxbro** [comment]
To only change max_zooms of OSM, we can do something like this:

``` diff
diff --git a/geoviews/plotting/bokeh/__init__.py b/geoviews/plotting/bokeh/__init__.py
index 6a2c16c..6956152 100644
--- a/geoviews/plotting/bokeh/__init__.py
+++ b/geoviews/plotting/bokeh/__init__.py
@@ -49,7 +49,7 @@ from ...operation import (
     project_quadmesh,
     project_vectorfield,
 )
-from ...tile_sources import _ATTRIBUTIONS
+from ...tile_sources import _ATTRIBUTIONS, OSM
 from ...util import line_types, poly_types
 from . import callbacks  # noqa
 from .plot import GeoOverlayPlot, GeoPlot
@@ -60,6 +60,9 @@ except ImportError:
     class ImageStackPlot:
         ...
 
+_TILE_MAX_ZOOM = {OSM: 19}
+
+
 class TilePlot(GeoPlot):
 
     style_opts = [
@@ -102,6 +105,8 @@ class TilePlot(GeoPlot):
                                 'Must contain {X}/{Y}/{Z}, {XMIN}/{XMAX}/{YMIN}/{YMAX} '
                                 'or {Q} template strings.')
             params = {'url': element.data}
+            if element in _TILE_MAX_ZOOM:
+                params['max_zoom'] = _TILE_MAX_ZOOM[element]
             for zoom in ('min_zoom', 'max_zoom'):
                 if zoom in style:
                     params[zoom] = style[zoom]
```

File: examples/user_guide/Working_with_Bokeh.ipynb [hunk 1]
```diff
@@ -95,6 +95,22 @@
     "gv.util.get_tile_rgb(tiles, bbox=(-180, -70, 180, 70), zoom_level=1).opts(width=500, height=500, projection=ccrs.Orthographic())"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note, that the `max_zoom` default is set to 19 for all tile sources; if you need a more detailed map, you can increase this value. However, if the tile source does not support the requested zoom level, the tile source could return a blank map instead, depending on how the tile source handles those requests."
```
Comment (ahuang11) [inline examples/user_guide/Working_with_Bokeh.ipynb:102]:
```suggestion
    "Note, that a `max_zoom` default is set for some tile sources; if you need a more detailed map, and the tile source supports it, you can increase this value. However, if the tile source does not support the requested zoom level, the tile source could return tiles that are empty or tiles that note "map data not yet available", depending on how the tile source handles those requests."
```

File: examples/user_guide/Working_with_Bokeh.ipynb [hunk 2]
```diff
@@ -95,6 +95,22 @@
     "gv.util.get_tile_rgb(tiles, bbox=(-180, -70, 180, 70), zoom_level=1).opts(width=500, height=500, projection=ccrs.Orthographic())"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note, that a `max_zoom` default is set for some tile sources; if you need a more detailed map, and the tile source supports it, you can increase this value. However, if the tile source does not support the requested zoom level, the tile source could return tiles that are empty or tiles that note "map data not yet available", depending on how the tile source handles those requests."
```
Comment (hoxbro) [inline examples/user_guide/Working_with_Bokeh.ipynb:102]:
```suggestion
    "Note, that a `max_zoom` default is set for some tile sources; if you need a more detailed map, and the tile source supports it, you can increase this value. However, if the tile source does not support the requested zoom level, the tile source could return tiles that are empty or tiles that note \"map data not yet available\", depending on how the tile source handles those requests."
```

**Review by hoxbro — APPROVED** [review]

**Comment by ahuang11** [comment]
Test will fail due to rc mismatch, but tested the new test and it's ready to merge (I don't have permissions) and added to the changelog.

**Pull request merged**